plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/editors/groovy/GroovyJob.java

```diff
@@ -157,28 +121,19 @@ public void run() {
             try {
                 shell.run(script,name, new String[] {});
                 String outputStream = shell.getProperty("out").toString();
+                GroovyConsoleContent.writeFileNameIntoConsole(name);
                 GroovyConsoleContent.writeIntoConsole(outputStream, true);
                 GroovyConsoleContent.writeIntoConsole("END");
                 status = IStatus.OK;
-                closeOutputPrintWriter();
-            } catch (MissingPropertyException e){
-                e.printStackTrace(pw);
-                String sStackTrace = sw.toString();
-                GroovyConsoleContent.writeIntoConsole(sStackTrace);
-                LocalDateTime now = LocalDateTime.now();
-                outputPrintWriter.write("\n" + dtf.format(now) + "\n" + sStackTrace);
-                GroovyConsoleContent.writeIntoConsole("BAD_END");   
-                status = IStatus.ERROR;
-                closeOutputPrintWriter();
-            } catch(Exception e){
+            } catch (Exception e){
                 e.printStackTrace(pw);
                 String sStackTrace = sw.toString();
+            	GroovyConsoleContent.writeFileNameIntoConsole(name);
                 GroovyConsoleContent.writeIntoConsole(sStackTrace);
-                LocalDateTime now = LocalDateTime.now();
-                outputPrintWriter.write("\n" + dtf.format(now) + "\n" + sStackTrace);
-                GroovyConsoleContent.writeIntoConsole("BAD_END");   
+                GroovyConsoleContent.writeIntoConsole("BAD_END");
+                System.err.println("Error in the groovy file");
```

Comment:
This error must be printed in groovyconsole and log file

Reply:
I don't really understand,
I have to write something like GroovyConsoleContent.writeIntoConsole(System.err.println("Error in the groovy file"))
and
LOGGER.trace(System.err.println("Error in the groovy file")) ?

I searched how to retrieve the "stderr" from binding like I did for "out" but I didn't find.

Reply:
As @ebocher said, you have to print your message ( which you have collected by replacing groovy out stream by yours ) in two output :
- In the groovy console UI, you already have this with the lines : 
```groovy
GroovyConsoleContent.writeFileNameIntoConsole(name);
GroovyConsoleContent.writeIntoConsole(sStackTrace);
```
- In the RCP application (in our case DBeaver) logging platform, using an instance of `org.orbisgis.core.logger.Logger`, and you already have one declared in this class. You will print the result (success or failure) of the script into the `error()` or the `info()` method, and you will print the body of the log (what the user print using the `println` call) into the `debug()` or `trace()`.